src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1075,7 +1075,12 @@ private static JsonWebToken ValidateSignature(string token, TokenValidationParam
             }
             else
             {
-                var key = JwtTokenUtilities.ResolveTokenSigningKey(jwtToken.Kid, jwtToken.X5t, validationParameters);
+                SecurityKey key;
+                if (validationParameters.ConfigurationManager == null)
+                    key = JwtTokenUtilities.ResolveTokenSigningKey(jwtToken.Kid, jwtToken.X5t, validationParameters);
+                else
+                    key = JwtTokenUtilities.ResolveTokenSigningKeyWithConfig(jwtToken.Kid, jwtToken.X5t, validationParameters);
```

Comment:
I've separated this out into two different methods in the hopes that this will make it easier for us to add a ValidateAsync(...) API in the future.